test/test_holidays.rb

```diff
@@ -139,6 +139,12 @@ def test_any_region_holiday_next
     end
   end
 
+  def test_any_region_holiday_year
+    holidays = Holidays.year_holidays([:ca_on], Date.civil(2016, 2, 23))
+    holidays.each {|h| puts h}
+    # assert_equal 12, holidays.length
```

Comment:
In addition, I know that I am asking for a stretch, but can you please add a few additional tests? 

I'm trying to expand the general test coverage and I think that we should start covering edge cases for each piece of functionality.

Potential edge cases:

* Jan 1 (i.e. get all holidays for that year)
* Dec 31 (i.e. get the last holiday, or none if NYE is not considered a holiday in a particular region)
* A few random months/days in different parts of the year
* What happens when someone specifies Feb 29 on a non leap year?
* A few different years (1990, 2016, 2020, 2050, just random years)
* Call without a specified year (i.e. default of `Date.today`)
* Anything else I missed?

Each of these should be their own test. I'm not a huge fan of putting multiple test cases into one test function.

Again, I recognize that this is asking for a bit of a stretch and that we don't always do this for other tests but I think it A) wouldn't be that much additional effort and B) is the 'right' way to do things going forward.

Let me know what you think!